### PR TITLE
Move projectBasePath and coreBasePath

### PR DESCRIPTION
**Pull request by ArneGudermann**
This PR move coreBasePath and projectBasePath from the utils to  the config related to #570

### CODE REVIEW

File: core/config.py [hunk 1]
```diff
@@ -44,6 +45,11 @@ def __setitem__(self, key, value):
 __project_id = google.auth.default()[1]
 __version = os.getenv("GAE_VERSION")
 
+# Determine our basePath (as os.getCWD is broken on appengine)
+__project_base_path = str(Path().absolute())
+
+__core_base_path = globals()["__file__"].replace("/viur/core/config.py", "")
```
Comment (phorward) [inline core/config.py:51]: Direct use of `__file__` should work as well (please verify!)
```suggestion
__core_base_path = __file__.replace("/viur/core/config.py", "")
```

File: core/render/__init__.py [hunk 2]
```diff
@@ -4,16 +4,14 @@
...
-
+from viur.core import conf
 
 def isViAvailable():
-    from viur.core.utils import projectBasePath
-    return path.exists(path.join(projectBasePath, "vi", "main.html"))
+    return path.exists(path.join(conf["viur.instance.core_base_path"], "vi", "main.html"))
 
 
 def isAdminAvailable():
```
Comment (phorward) [inline core/render/__init__.py:13]: General question: Is this only a relict of the Pyodie-based QT-Admin-Test?

**Review by phorward — CHANGES_REQUESTED** [review]

File: core/config.py (hunk 1 above)
Comment (ArneGudermann) [reply core/config.py:51]: yes `__file__` works too.

File: core/render/__init__.py (hunk 2 above)
Comment (ArneGudermann) [reply core/render/__init__.py:13]: No it`s used here https://github.com/viur-framework/viur-core/blob/221743136e867911bfa0fa994ed9db4ae820609d/core/render/vi/__init__.py#L152-L155 in example.

Comment (phorward) [reply core/render/__init__.py:13]: Nevermind. We should kill this later on.

**Review by phorward — APPROVED** [review]

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: core/config.py [hunk 3]
```diff
@@ -44,6 +45,11 @@ def __setitem__(self, key, value):
 __project_id = google.auth.default()[1]
 __version = os.getenv("GAE_VERSION")
 
+# Determine our basePath (as os.getCWD is broken on appengine)
+__project_base_path = str(Path().absolute())
+
+__core_base_path = __file__.replace("/viur/core/config.py", "")
```
[the remark this answers is not in the document]
Comment (phorward) [reply core/config.py:51]: So Path implements `__str__`? Then it's ok.

[the remark this answers is not in the document]
Comment (phorward) [reply core/config.py:51]:
```
$ cat test.py 
import pathlib

p = pathlib.Path().absolute()
print(p + "lol")
$ python test.py 
Traceback (most recent call last):
  File "/tmp/test.py", line 4, in <module>
    print(p + "lol")
TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'
```
Doesn't work as I expected, anyway I don't think it breaks anything and we can make it so.

[the remark this answers is not in the document]
Comment (sveneberth) [reply core/config.py]: @phorward Yes, but not all binary operators. But it should in general work for sth. like `open()`

File: core/render/html/default.py [hunk 4]
```diff
@@ -101,13 +102,13 @@ def getTemplateFileName(self, template: str, ignoreStyle: bool = False) -> str:
                       template + ".html"]
         for fn in fnames:  # check subfolders
             prefix = template.split("_")[0]
-            if os.path.isfile(os.path.join(utils.projectBasePath, htmlpath, prefix, fn)):
+            if os.path.isfile(os.path.join(conf["viur.instance.core_base_path"], htmlpath, prefix, fn)):
```
Comment (sveneberth) [inline core/render/html/default.py:105]: This is not correctly substituted, it's the project path.

**Review by sveneberth — CHANGES_REQUESTED** [review]
There are some issues, see https://github.com/viur-framework/viur-core/pull/598

**Review by sveneberth — DISMISSED** [review]

**Comment by phorward** [comment]
@sveneberth are you happy with this now?

**Review by phorward — APPROVED** [review]

**Review by sveneberth — APPROVED** [review]

**Pull request merged**